core/encoding.go

```diff
@@ -38,6 +38,9 @@ type Encoder interface {
 	// VerifyChunks takes in the chunks, indices, commitments, and encoding parameters and returns an error if the chunks are invalid.
 	VerifyChunks(chunks []*Chunk, indices []ChunkNumber, commitments BlobCommitments, params EncodingParams) error
 
+	// VerifyBatch takes in
+	UniversalVerifyChunks(params EncodingParams, samples []Sample, numBlobs int) error
```

Comment:
It looks these samples belong to the same sub batch. In that case, it'll be more readability to pass in a SubBatch as it carries logical meaning for the parameters.